iree/compiler/InputConversion/Common/QuantizedMatmulToMatmul.cpp

```diff
@@ -0,0 +1,236 @@
+      indexOfRhsSumsInput = addInput(rhsSums, mapToColumnDim);
+      indexOfLhsZpInput = addInput(lhsZp, mapToNone);
+    }
+    if (!lhsZpIsConstantZero && !rhsZpIsConstantZero) {
+      Value lhsZpTimesRhsZp = rewriter.create<arith::MulIOp>(loc, lhsZp, rhsZp);
+      Value kSize = rewriter.create<arith::IndexCastOp>(
+          loc, accElTy, rewriter.create<tensor::DimOp>(loc, lhs, 1));
+      Value lhsZpTimesRhsZpTimesKSize =
+          rewriter.create<arith::MulIOp>(loc, lhsZpTimesRhsZp, kSize);
+      indexOfLhsZpTimesRhsZpTimesKSizeInput =
+          addInput(lhsZpTimesRhsZpTimesKSize, mapToNone);
+    }
+    // Add the indexing map for the initResult 'output' even though it's unused.
+    indexingMaps.push_back(mapIdentity);
+    // Create the generic putting all the terms together.
+    SmallVector<StringRef> iterators{"parallel", "parallel"};
+    rewriter.replaceOpWithNewOp<linalg::GenericOp>(
+        quantizedMatmulOp, acc.getType(), ins, ValueRange{initResult},
+        indexingMaps, iterators,
+        [=](OpBuilder &b, Location loc, ValueRange args) {
+          Value matmulEl = args[indexOfMatmulInput];
+          Value lhsSumsEl = args[indexOfLhsSumsInput];
+          Value rhsSumsEl = args[indexOfRhsSumsInput];
+          Value lhsZp = args[indexOfLhsZpInput];
+          Value rhsZp = args[indexOfRhsZpInput];
+          Value lhsZpTimesRhsZpTimesKSize =
+              args[indexOfLhsZpTimesRhsZpTimesKSizeInput];
+          Value result = matmulEl;
+          // If the rhs zero-point is not a constant zero, we need to add it
+          // times the sums along rows of lhs.
+          if (!rhsZpIsConstantZero) {
+            Value lhsSumsElTimesRhsZp =
+                b.create<arith::MulIOp>(loc, lhsSumsEl, rhsZp);
+            result = b.create<arith::SubIOp>(loc, result, lhsSumsElTimesRhsZp);
+          }
+          // If the lhs zero-point is not a constant zero, we need to add it
+          // times the sums along columns of rhs.
+          if (!lhsZpIsConstantZero) {
+            Value rhsSumsElTimesLhsZp =
+                b.create<arith::MulIOp>(loc, rhsSumsEl, lhsZp);
+            result = b.create<arith::SubIOp>(loc, result, rhsSumsElTimesLhsZp);
+          }
+          // Add the final correction term, if neither zero-point is cst zero.
+          if (!lhsZpIsConstantZero && !rhsZpIsConstantZero) {
+            result =
+                b.create<arith::AddIOp>(loc, result, lhsZpTimesRhsZpTimesKSize);
+          }
+          b.create<linalg::YieldOp>(loc, result);
+        });
+    return success();
+  }
+};
+
+/// Pass that lowers quantized_matmul to matmul.
+struct LinalgQuantizedMatmulToMatmulPass
+    : public LinalgQuantizedMatmulToMatmulPassBase<
+          LinalgQuantizedMatmulToMatmulPass> {
+  void runOnOperation() override {
+    Operation *op = getOperation();
+    MLIRContext *context = op->getContext();
+    RewritePatternSet patterns(context);
+    patterns.add<QuantizedMatmulToMatmul>(context);
```

Comment:
Since this pattern is creating `tensor.dim` of the result of the `linalg.quantized_matmul`, can you add this pattern as well to the pattern set : https://github.com/llvm/llvm-project/blob/f978497d67c832367af57da5541eedefcac6e7ac/mlir/include/mlir/Dialect/MemRef/Transforms/Passes.h#L46 .

This will resolve the `tensor.dim` in terms of its operands making the op truly dead.

Reply:
Thanks, done!